### PR TITLE
UIF-99 add modal form for create budget

### PR DESCRIPTION
**Pull request by Feboll**
## Purpose

Ability to create summary information for the budget record
https://issues.folio.org/browse/UIF-99

## Approach

Add modal for create budget

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          fundId: fund.id,
+          name: `${fund.code}-${fiscalYear.code}`,
+        });
+        const { name, id } = budget;
+        showCallout('ui-finance.budget.hasBeenCreated', 'success', { name, fundName: fund.name });
+        const path = `/finance/budget/${id}/view`;
+
+        history.push(path);
+      } catch (e) {
+        console.log(e);
```
Comment (AndreiShumski2) [inline src/components/Fund/BudgetAddModal/BudgetAddModal.js]: remove, please

Comment (Feboll) [reply src/components/Fund/BudgetAddModal/BudgetAddModal.js]: fixed

File: src/components/Fund/FundDetails/FundDetailsContainer.js [hunk 2]
```diff
@@ -161,6 +163,17 @@ const FundDetailsContainer = ({
     [],
   );
 
+  const addBudgetButton = useCallback((status, count) => {
+    return count === 0 && (
```
Comment (AndreiShumski2) [inline src/components/Fund/FundDetails/FundDetailsContainer.js]: `(!count)` ?

Comment (Feboll) [reply src/components/Fund/FundDetails/FundDetailsContainer.js]: changed

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import BudgetAddModalForm from './BudgetAddModalForm';
+import {
+  budgetResource,
+  fiscalYearsResource
+} from '../../../common/resources';
+import getFiscalYearsForSelect from '../../../Utils/getFiscalYearsForSelect';
+import { BUDGET_STATUSES } from '../../Budget/constants';
+
+
+const BudgetAddModal = ({ history, mutator, resources, onClose, fund, budgetStatus }) => {
```
Comment (NikitaSedyx) [inline src/components/Fund/BudgetAddModal/BudgetAddModal.js]: AddBudgetModal?

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
```
Comment (NikitaSedyx) [inline src/components/Fund/BudgetAddModal/BudgetAddModal.js]: why is it placed under fund?

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        const { name, id } = budget;
+        showCallout('ui-finance.budget.hasBeenCreated', 'success', { name, fundName: fund.name });
+        const path = `/finance/budget/${id}/view`;
+
+        history.push(path);
+      } catch (e) {
+        showCallout('ui-finance.budget.hasNotBeenCreated', 'error');
+      }
+    },
+    [history, fiscalYears, fund, mutator]
```
Comment (NikitaSedyx) [inline src/components/Fund/BudgetAddModal/BudgetAddModal.js]: `fiscalYears` is always new array

File: src/components/Fund/FundDetails/FundDetailsContainer.js [hunk 6]
```diff
@@ -161,6 +163,17 @@ const FundDetailsContainer = ({
     [],
   );
 
+  const addBudgetButton = useCallback((status, count) => {
+    return !count && (
+      <Button
+        data-test-add-budget-button
+        onClick={() => toggleAddBudget(status)}
```
Comment (NikitaSedyx) [inline src/components/Fund/FundDetails/FundDetailsContainer.js]: status? your state is boolean, plus we have a hook for modal toggle

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js (hunk 3 above)
Comment (Feboll) [reply src/components/Fund/BudgetAddModal/BudgetAddModal.js]: renamed

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js (hunk 4 above)
Comment (Feboll) [reply src/components/Fund/BudgetAddModal/BudgetAddModal.js]: Moved to Budget folder

File: src/components/Fund/BudgetAddModal/BudgetAddModal.js (hunk 5 above)
Comment (Feboll) [reply src/components/Fund/BudgetAddModal/BudgetAddModal.js]: I tried memoized `fiscalYears`. Take a look my changes.

File: src/components/Fund/FundDetails/FundDetailsContainer.js (hunk 6 above)
Comment (Feboll) [reply src/components/Fund/FundDetails/FundDetailsContainer.js]: Changed name for variable

**Pull request merged**